src/client/kotlin/tech/thatgravyboat/skyblockapi/api/profile/effects/EffectsAPI.kt

```diff
@@ -0,0 +1,120 @@
+package tech.thatgravyboat.skyblockapi.api.profile.effects
+
+import kotlinx.datetime.Clock.System
+import kotlinx.datetime.Instant
+import tech.thatgravyboat.skyblockapi.api.data.stored.EffectsStorage
+import tech.thatgravyboat.skyblockapi.api.events.base.Subscription
+import tech.thatgravyboat.skyblockapi.api.events.base.predicates.OnlyWidget
+import tech.thatgravyboat.skyblockapi.api.events.chat.ChatReceivedEvent
+import tech.thatgravyboat.skyblockapi.api.events.info.TabListHeaderFooterChangeEvent
+import tech.thatgravyboat.skyblockapi.api.events.info.TabWidget
+import tech.thatgravyboat.skyblockapi.api.events.info.TabWidgetChangeEvent
+import tech.thatgravyboat.skyblockapi.api.events.misc.RegisterCommandsEvent
+import tech.thatgravyboat.skyblockapi.api.events.screen.ContainerInitializedEvent
+import tech.thatgravyboat.skyblockapi.api.profile.community.CommunityCenterAPI.cookieAteRegex
+import tech.thatgravyboat.skyblockapi.helpers.McClient
+import tech.thatgravyboat.skyblockapi.modules.Module
+import tech.thatgravyboat.skyblockapi.utils.extentions.cleanName
+import tech.thatgravyboat.skyblockapi.utils.extentions.getRawLore
+import tech.thatgravyboat.skyblockapi.utils.extentions.parseDuration
+import tech.thatgravyboat.skyblockapi.utils.extentions.parseWordDuration
+import tech.thatgravyboat.skyblockapi.utils.regex.RegexGroup
+import tech.thatgravyboat.skyblockapi.utils.regex.RegexUtils.anyMatch
+import tech.thatgravyboat.skyblockapi.utils.regex.RegexUtils.contains
+import tech.thatgravyboat.skyblockapi.utils.text.Text
+import tech.thatgravyboat.skyblockapi.utils.text.Text.send
+import tech.thatgravyboat.skyblockapi.utils.text.TextColor
+import tech.thatgravyboat.skyblockapi.utils.text.TextStyle.color
+import tech.thatgravyboat.skyblockapi.utils.time.until
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.days
+
+@Module
+object EffectsAPI {
+
+    val boosterCookieExpireTime get() = EffectsStorage.boosterCookieExpireTime
+
+
+    @Subscription
+    fun onChat(event: ChatReceivedEvent) {
+        if (cookieAteRegex.contains(event.text)) {
+            updateBoosterCookieExpireTime(boosterCookieExpireTime.until() + 4.days)
+        }
+    }
+
+    @Subscription
+    fun onInventoryFullyLoaded(event: ContainerInitializedEvent) {
+        if (event.title == "SkyBlock Menu") {
+            val cookieLore = event.itemStacks.find { it.cleanName == "Booster Cookie" }?.getRawLore() ?: return
+            cookieInventoryRegex.anyMatch(cookieLore, "duration") { (duration) ->
+                val parsedDuration = duration.parseDuration() ?: return@anyMatch
+                updateBoosterCookieExpireTime(parsedDuration)
+            }
+        }
+    }
+
+    @Subscription
+    fun onTabFooterUpdate(event: TabListHeaderFooterChangeEvent) {
+        val asshole = event.newFooterChunked.find { "Cookie Buff" in it } ?: return
```

Comment:
change name :3

Reply:
also fix that this is not checking size bounds, will also need to be updated for when chunks are trimmed